scipy/stats/mstats_extras.py

```diff
@@ -161,13 +161,12 @@ def _hdsd_1D(data, prob):
 
         vv = np.arange(n) / float(n-1)
         betacdf = beta.cdf
-
+        
         for (i,p) in enumerate(prob):
             _w = betacdf(vv, (n+1)*p, (n+1)*(1-p))
             w = _w[1:] - _w[:-1]
```

Comment:
This looks like `w = -np.diff(_w)` to me.

Reply:
It's `w = np.diff(_w)` (note the sign), but not sure it'd improve speed.

Reply:
It is first taking the shifted array then the original order so it is negative

Reply:
Check this small example:

```
import numpy as np
_w = np.array([1, 2, 3, 8, 1, 3])
_w[1:] - _w[:-1]
array([ 1,  1,  5, -7,  2])
np.diff(_w)
array([ 1,  1,  5, -7,  2])
```